src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs

```diff
@@ -50,83 +52,172 @@ static IdentityModelEventSource()
         /// </summary>
         public static IdentityModelEventSource Logger { get; }
 
+        [Event(6, Level = EventLevel.LogAlways)]
+        public void WriteAlways(string message)
+        {
+            if (IsEnabled())
+            {
+                message = PrepareMessage(EventLevel.LogAlways, message);
+                WriteEvent(6, message);
+            }
+        }
+
+        [NonEvent]
+        public void WriteAlways(string message, params object[] args)
+        {
+            if (IsEnabled())
+            {
+                if (args != null)
+                    WriteAlways(string.Format(CultureInfo.InvariantCulture, message, args));
+                else
+                    WriteAlways(message);
+            }
+        }
+
         [Event(1, Level = EventLevel.Verbose)]
         public void WriteVerbose(string message)
         {
-            if (IsEnabled() && _logLevel >= EventLevel.Verbose)
+            if (IsEnabled() && LogLevel >= EventLevel.Verbose)
             {
-                message = PrepareMessage(message, EventLevel.Verbose);
+                message = PrepareMessage(EventLevel.Verbose, message);
                 WriteEvent(1, message);
             }
         }
 
+        [NonEvent]
+        public void WriteVerbose(string message, params object[] args)
```

Comment:
@tushargupta51 I though so also. However, the 'Event' attribute imposes a very strict API.